File: PART-B/Webview/app/src/main/java/com/example/webview/MainActivity.java
package com.example.webview;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {
Button b1;
Button b2;
 EditText e1;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        b1=findViewById(R.id.b1);
        b2=findViewById(R.id.b2);
        e1=findViewById(R.id.e1);
        b1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent i = new Intent(MainActivity.this, MainActivity2.class);
                i.putExtra("load",e1.getText().toString());
                startActivity(i);
            }
        });
        b2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent i= new Intent(MainActivity.this, MainActivity2.class);
                i.putExtra("load","default12");
                startActivity(i);
            }
        });

    }
}

File: MyApplication/app/src/main/java/com/example/myapplication/MainActivity.kt
package com.example.myapplication

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.example.myapplication.databinding.ActivityMainBinding
import com.google.firebase.firestore.ktx.firestore
import com.google.firebase.ktx.Firebase
import kotlin.random.Random
import kotlin.random.nextInt

class MainActivity : AppCompatActivity() {
    lateinit var binding:ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding= ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)


        binding.plyOfflineBtn.setOnClickListener {
createOfflineGame()
        }
binding.createOnlineGameBtn.setOnClickListener {
    createOnlineGame()
}
binding.joinOnlineGameBtn.setOnClickListener {
    joinOnlineGame()
}
    }

    fun createOfflineGame()
    {
        GameData.saveGameModel(
            GameModel(
                gamestatus = GameStatus.JOINED
            )
        )
        startGame()
    }

    fun createOnlineGame()
    {
GameData.myID="X"
        GameData.saveGameModel(
            GameModel(
                gamestatus = GameStatus.CREATED,
                gameId =  Random.nextInt(1000..9999).toString()
            )
        )
        startGame()
    }

    fun joinOnlineGame() {
        val gameId = binding.gameIdInput.text.toString()
        if (gameId.isEmpty()) {
            binding.gameIdInput.setError("Please Enter Game Id")
            return
        }
        GameData.myID = "O"
        Firebase.firestore.collection("games")
            .document(gameId)
            .get()
            .addOnSuccessListener { documentSnapshot ->
                val model = documentSnapshot.toObject(GameModel::class.java)
                if (model == null) {
                    binding.gameIdInput.setError("Please Enter a Valid GameId")
                } else {
                    if (model.gamestatus == GameStatus.CREATED) {
                        // If the game is still in CREATED status, then the second player can join
                        model.gamestatus = GameStatus.JOINED
                        GameData.saveGameModel(model)
                        startGame()
                    } else {
                        // If the game is already joined by another player, show an error
                        binding.gameIdInput.setError("Game is already joined by another player")
                    }
                }
            }
    }

    fun startGame(){
            startActivity(Intent(this,GameActivity::class.java))
        }


}